### PR TITLE
Feature/connect stellar actions

### PR DESCRIPTION
**Pull request by jalarg**
# Summary
- Add handlers to use the functions of stellar, this functions include: Send Transaction, Load Balance, History of payments, minimum Balance and Albedo login.
- Add HOC to control user Authentication to log into the /wallet section of the app.
- Refactor cypress testing to include redux usage to log into /wallet

# Details
- Introduces Stellar functionality, enabling Send Transactions, Balance Loading, Payment History, Minimum Balance checks, and Albedo login handlers.

# Evidence 

## Stellar wallet: 

Home UI

wallet UI

Albedo

### CODE REVIEW

File: src/actions/stellar/albedoLogin.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import albedo from "@albedo-link/intent";
+import { useDispatch } from "react-redux";
+import { login } from "../../globalRedux/store";
+
+function albedoLogin(dispatch = useDispatch()) {
```
Comment (Genebson) [inline src/actions/stellar/albedoLogin.ts:5]: This fuction is missing the infinitive verb at the beginning.

File: src/actions/utils/roundNumbers.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const roundNumber = (numberString: string) => {
```
Comment (Genebson) [inline src/actions/utils/roundNumbers.ts]: We should decide which type of functions we're gonna use.

File: src/actions/wallets/albedoWallet.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  publicKey: string;
+  secretKey: string;
+}
+
+interface IAlbedo {
+  destinationId: string;
+  amount: string;
+}
+
+export default class Albedo extends PrivateKeyWallet {
```
Comment (Genebson) [inline src/actions/wallets/albedoWallet.ts]: Why does it extend from PrivateKey? It doesn't have to depend on another wallet.

File: src/actions/wallets/albedoWallet.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    super({ publicKey, secretKey });
+  }
+  async sendTransaction({ destinationId, amount }: IAlbedo): Promise<any> {
+    const result = await albedo.pay({
+      destination: destinationId,
+      amount,
+      network: "testnet",
+      submit: true,
+    });
+    return result;
```
Comment (Genebson) [inline src/actions/wallets/albedoWallet.ts:54]:
```suggestion
    return await albedo.pay({
      destination: destinationId,
      amount,
      network: "testnet",
      submit: true,
    });
```

File: src/actions/wallets/albedoWallet.ts (hunk 3 above)
Comment (Genebson) [inline src/actions/wallets/albedoWallet.ts]: I think this class should have the login method as well.

File: src/actions/wallets/walletSwitcher.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+export default class WalletSwitcher {
+  static createWallet(authCredentials: IWalletSwitcher) {
+    const { walletType, publicKey, secretKey } = authCredentials;
+    switch (walletType) {
+      case "privateKey":
+        return new PrivateKeyWallet({ publicKey, secretKey });
+      case "albedo":
+        return new AlbedoWallet({ publicKey, secretKey });
+      default:
+        throw new Error("Tipo de wallet no válido");
```
Comment (Genebson) [inline src/actions/wallets/walletSwitcher.ts]: English, please.

File: src/components/Navbar.tsx [hunk 6]
```diff
@@ -2,11 +2,17 @@ import Link from "next/link";
...
+import logout from "../assets/logout.png";
 import { Avatar } from "antd";
 import { FaRegMoon } from "react-icons/fa";
 import { FiSun } from "react-icons/fi";
+import { useSelector, useDispatch } from "react-redux";
+import { IAuth } from "../types/types";
+import { handleLogout } from "../actions/handlers";
 
 const Navbar = () => {
+  const dispatch = useDispatch();
```
Comment (Genebson) [inline src/components/Navbar.tsx:14]: Type this component, please.

File: src/actions/wallets/privateKeyWallet.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+              console.error("Error in payment stream");
+              reject(error);
+            },
+          });
+      } catch (err) {
+        console.error("ERROR!", err);
+        reject(err);
+      }
+    });
+  }
```
Comment (Genebson) [inline src/actions/wallets/privateKeyWallet.ts]: I think this doesn't belong here.

Comment (Genebson) [reply src/actions/wallets/privateKeyWallet.ts]: This wallet should have just the login and the payment

File: src/components/Footer.tsx [hunk 8]
```diff
@@ -5,7 +5,7 @@ import { PiCopyrightBold } from "react-icons/pi";
 
 const Footer = () => {
```
Comment (Genebson) [inline src/components/Footer.tsx]: Same here.

File: src/components/Payments.tsx [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  {
+    title: "Amount",
+    dataIndex: "amount",
+    render: (amount: string) => roundNumber(amount),
+    width: 100,
+  },
+];
+
+const Payments: React.FC<IPayments> = ({ payments }) => {
+  const paymentsHistory = payments.map((payment, index) => {
```
Comment (Genebson) [inline src/components/Payments.tsx:34]: You can destructure payment here.

File: src/actions/utils/roundNumbers.ts (hunk 2 above)
Comment (jalarg) [reply src/actions/utils/roundNumbers.ts]: Solved in commit e8e88ff64e2c75343def6c68b7a14c22d053e5d8.

File: src/actions/wallets/albedoWallet.ts (hunk 3 above)
Comment (jalarg) [reply src/actions/wallets/albedoWallet.ts]: issue resolved.

File: src/actions/handlers/handleCloseModal.ts [hunk 10]
```diff
@@ -1,6 +1,6 @@
-import { ICloseModal } from "../types/types";
+import { ICloseModal } from "../../types/types";
 
-function closeModalHandler({
+async function handleCloseModal({
```
Comment (Genebson) [inline src/actions/handlers/handleCloseModal.ts:3]: This is async but we're not awaiting.

File: src/actions/handlers/handleLogin.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    }
+    const wallet = WalletSwitcher.createWallet({
+      walletType: "privateKey",
+      publicKey: (publicKey) ? publicKey : Keypair.fromSecret(secretKey).publicKey(),
+      secretKey,
+    });
+    await wallet.login({ dispatch, secretKey });
+    message.success("Login successful!");
+  } catch (error) {
+    console.log("Error:", error);
```
Comment (Genebson) [inline src/actions/handlers/handleLogin.ts:24]: Use `console.error` here

File: src/actions/handlers/handleLoginAlbedo.ts [hunk 12]
```diff
@@ -0,0 +1,23 @@
+import { message } from "antd";
+import WalletSwitcher from "../wallets/walletSwitcher";
+
+interface IHandlerAlbedo {
+  dispatch: (action: any) => void;
+}
+
+export default handleLoginAlbedo;
```
Comment (Genebson) [inline src/actions/handlers/handleLoginAlbedo.ts:8]: This should be at the bottom of the file.

File: src/actions/handlers/handleLoginAlbedo.ts [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+async function handleLoginAlbedo({ dispatch }: IHandlerAlbedo) {
```
Comment (Genebson) [inline src/actions/handlers/handleLoginAlbedo.ts:10]:
```suggestion
async function handleAlbedoLogin({ dispatch }: IHandlerAlbedo) {
```

File: src/actions/handlers/handleLogout.ts [hunk 14]
```diff
@@ -0,0 +1,13 @@
+import { logout } from "../../globalRedux/store";
+import { message } from "antd";
+
+function handleLogout(dispatch: any) {
+  try {
+    dispatch(logout());
+    message.success("Logout Successful");
+  } catch (error) {
+    console.log("Error:", error);
```
Comment (Genebson) [inline src/actions/handlers/handleLogout.ts:9]: Same here.

File: src/actions/handlers/handleMinimumBalance.ts [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+
+async function handleMinimumBalance({
+  publicKey,
+  setBalance,
+  setIsLoading,
+}: IAddMiniumBalanceHandler) {
+  if (publicKey) {
+    setIsLoading(true);
+    await addMinimumBalance(publicKey);
+    const balance = await checkBalance(publicKey);
```
Comment (Genebson) [inline src/actions/handlers/handleMinimumBalance.ts:14]: You can destructure balance here.

File: src/actions/handlers/handleWalletInformation.ts [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+async function handleWalletInformation({
+  setIsLoading,
+  setBalance,
+  setPayments,
+  publicKey,
+}: IHandleWalletInformationProps) {
+  if (publicKey) {
+    setIsLoading(true);
+    try {
+      const balance = await checkBalance(publicKey);
```
Comment (Genebson) [inline src/actions/handlers/handleWalletInformation.ts:13]: You can destructure here as well.

**Pull request merged**